rust/src/storage/s3/dynamodb_lock.rs

```diff
@@ -154,6 +154,12 @@ pub enum DynamoError {
     #[error("Could not acquire lock for {0} sec")]
     TimedOut(u64),
 
+    /// Error returned by [`DynamoDbLockClient::acquire_lock`] which indicates that the lock could
+    /// not be acquired because the `is_non_acquirable` is set to `true`.
+    /// Usually this is done intentionally outside of [`DynamoDbLockClient`].
```

Comment:
I recommend adding the drop table example you had in the PR description to help readers better understand the use-case.

Reply:
sure, done